ir/src/symbol_table/mod.rs

```diff
@@ -250,34 +253,20 @@ impl SymbolTable {
     /// Returns an error if:
     /// - the identifier was not in the symbol table.
     /// - the identifier was not declared as a trace column binding.
+    /// TODO: update docs
     pub(crate) fn get_trace_access_by_name(
         &self,
         trace_access: &NamedTraceAccess,
     ) -> Result<IndexedTraceAccess, SemanticError> {
-        let elem_type = self.get_type(trace_access.name())?;
-        match elem_type {
-            IdentifierType::TraceColumns(columns) => {
-                if trace_access.idx() >= columns.size() {
-                    return Err(SemanticError::IndexOutOfRange(format!(
-                        "Out-of-range index '{}' while accessing named trace column group '{}' of length {}",
-                        trace_access.idx(),
-                        trace_access.name(),
-                        columns.size()
-                    )));
-                }
-
-                Ok(IndexedTraceAccess::new(
-                    columns.trace_segment(),
-                    columns.offset() + trace_access.idx(),
-                    trace_access.row_offset(),
-                ))
-            }
-            _ => Err(SemanticError::InvalidUsage(format!(
-                "Identifier {} was declared as a {} not as a trace column",
-                trace_access.name(),
-                elem_type
-            ))),
-        }
+        let symbol_type = self.get_type(trace_access.name())?;
+        trace_access.validate(symbol_type)?;
+
+        let IdentifierType::TraceColumns(columns) = symbol_type else { panic!("symbol table validation of named trace access failed.") };
```

Comment:
nit: Any reason to panic here instead of returning an error?

Reply:
Yes - the reason I did this was that I want to move all the errors to the validation. This error is already handled in the `validate` call above, so the else case here should never happen. If it does happen, then something has gone badly wrong with the validation. I could put this into an `unreachable` instead for clarity if you think that's better

Reply:
Thanks, I thought that could have been the reasoning. I don't have a strong opinion but to me `unreachable` does seem clearer for these cases but I'll defer to you.